contracts/Optimism_SpokePool.sol

```diff
@@ -0,0 +1,65 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@eth-optimism/contracts/libraries/bridge/CrossDomainEnabled.sol";
+import "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+import "./SpokePool.sol";
+
+/**
+ * @notice OVM specific SpokePool.
+ * @dev Uses OVM cross-domain-enabled logic for access control.
+ */
+
+contract Optimism_SpokePool is CrossDomainEnabled, SpokePool {
+    // Address of the L1 contract that acts as the owner of this SpokePool.
+    address public crossDomainAdmin;
+
+    event SetXDomainAdmin(address indexed newAdmin);
+
+    constructor(
+        address _crossDomainAdmin,
+        address _wethAddress,
+        uint64 _depositQuoteTimeBuffer,
+        address timerAddress
+    )
+        CrossDomainEnabled(Lib_PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER)
+        SpokePool(_wethAddress, _depositQuoteTimeBuffer, timerAddress)
+    {
+        _setCrossDomainAdmin(_crossDomainAdmin);
+    }
+
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
+
+    // TODO:
+    function setEnableRoute(
+        address originToken,
+        uint256 destinationChainId,
+        bool enable
+    ) public onlyFromCrossDomainAccount(crossDomainAdmin) {
+        _setEnableRoute(originToken, destinationChainId, enable);
+    }
+
+    // TODO:
+    function setDepositQuoteTimeBuffer(uint64 buffer) public onlyFromCrossDomainAccount(crossDomainAdmin) {
+        _setDepositQuoteTimeBuffer(buffer);
+    }
+
+    function initializeRelayerRefund(bytes32 relayerRepaymentDistributionProof)
+        public
+        onlyFromCrossDomainAccount(crossDomainAdmin)
+    {
+        _initializeRelayerRefund(relayerRepaymentDistributionProof);
+    }
+
+    /**************************************
+     *         INTERNAL FUNCTIONS         *
+     **************************************/
+
+    function _setCrossDomainAdmin(address newCrossDomainAdmin) internal {
```

Comment:
Is there any way to call this function after construction? Is this something we'd like to be able to change? (genuine question, not suggesting a change)

Reply:
there is a `setCrossDomainAdmin` that can call this after construction enabling us to change it. I think we'd want to change it in the event something broke within the L1<->L2 logic. one problem, though, that I can see with this pattern is it assumes that the cross domain owner implements sufficient methods to call `setCrossDomainAdmin` over the bridge. this should be doable as this is going to be sitting within the "Adapter" logic that I've created for the L1->L2 calls. We can create a base contract (`CrossDomainAdminBase`) or something equivalent that these adapters will inherit from. can think more on this in the next few PRs.

Reply:
Yeah i added in this function without telling you. We should allow this to be changed from L1

Reply:
Sorry, I'm a little confused. Are there two functions that allow you to `setCrossDomainAdmin`? If so, can we consolidate them into a single function?

Reply:
No there's only one, implemented in this contract. `setCrossDomainAdmin` is included in the `SpokePoolInterface`